File: Java/problem1417.java
class Solution {
    public String reformat(String s) {
        char[] chars = s.toCharArray();
        char[] strs = new char[(chars.length + 1) / 2];
        char[] nums = new char[(chars.length + 1) / 2];
        int strIdx = 0, numIdx = 0;
        for (char c : chars) {
            if (c >= 'a' && c <= 'z') {
                if (strIdx >= strs.length) {
                    return "";
                }
                strs[strIdx++] = c;
            }
            else {
                if (numIdx >= strs.length) {
                    return "";
                }
                nums[numIdx++] = c;
            }
        }
        char[] result = new char[strIdx + numIdx];
        if (strIdx >= numIdx) {
            for (int i = 0; i < strIdx; ++i) {
                result[2 * i] = strs[i];
                
            }
            for (int i = 0; i < numIdx; ++i) {
                result[2 * i + 1] = nums[i];
            }
        }
        else {
            for (int i = 0; i < numIdx; ++i) {
                result[2 * i] = nums[i];
                
            }
            for (int i = 0; i < strIdx; ++i) {
                result[2 * i + 1] = strs[i];
            }
        }
        return new String(result);
    }
}

File: python/problem1417.py
class Solution:
    def reformat(self, s: str) -> str:
        nums = list(filter(lambda x : x.isdigit(), s))
        letters = list(filter(lambda x : x.isalpha(), s))
        numLen = len(nums)
        letterLen = len(letters)
        if abs(numLen - letterLen) > 1:
            return ''
        result = ''
        if numLen < letterLen:
            for i in range(numLen):
                result += letters[i]
                result += nums[i]
            result += letters[-1]
        else:
            for i in range(letterLen):
                result += nums[i]
                result += letters[i]
            if numLen > letterLen:
                result += nums[-1]
        return result
